src/core/src/pass/serialize.cpp

```diff
@@ -420,6 +421,38 @@ class XmlSerializer : public ov::AttributeVisitor {
         }
     }
 
+    void pack_string_tensor(ov::AttributeAdapter<std::shared_ptr<ov::StringAlignedBuffer>>* adapter,
+                            std::shared_ptr<uint8_t>& data,
+                            size_t& total_size) {
+        // packed format is the following:
+        // <num_string>, <1st string offset>,..., <nth string offset>, <1st string raw format>,..., <nth rawformat>
+        auto num_elements = adapter->get()->get_num_elements();
+        auto strings = reinterpret_cast<std::string*>(adapter->get()->get_ptr());
+
+        // first run over all elements: calculate total memory required to hold all strings
+        size_t symbols_size =
+            std::accumulate(strings, strings + num_elements, size_t(0), [](size_t accum, const std::string str_ptr) {
+                return accum + str_ptr.size();
+            });
+        total_size = 4 * (1 + 1 + num_elements) + symbols_size;
+        data = std::shared_ptr<uint8_t>(new uint8_t[total_size], std::default_delete<uint8_t[]>());
+
+        int32_t* pindices = reinterpret_cast<int32_t*>(data.get());
+        pindices[0] = int32_t(num_elements);
+        pindices[1] = 0;
+        pindices += 2;
+        char* psymbols = reinterpret_cast<char*>(pindices + num_elements);
+        size_t current_symbols_pos = 0;
+
+        for (size_t ind = 0; ind < num_elements; ++ind) {
+            auto str = strings[ind];
+            psymbols = std::copy(str.begin(), str.end(), psymbols);
```

Comment:
do we need to concat all string together? I think we can write string one by one to a final place via `m_constant_write_handler` instead of concatenating them into a temp buffer

Reply:
done, I minimized memory consumption during serialization.